private/bufpkg/bufplugin/bufplugin.go

```diff
@@ -239,10 +244,78 @@ func ProtoRegistryConfigToPluginRegistry(config *registryv1alpha1.RegistryConfig
 			return nil, err
 		}
 		registryConfig.Maven = mavenConfig
+	} else if protoSwiftConfig := config.GetSwiftConfig(); protoSwiftConfig != nil {
+		swiftConfig, err := ProtoSwiftConfigToSwiftRegistryConfig(protoSwiftConfig)
+		if err != nil {
+			return nil, err
+		}
+		registryConfig.Swift = swiftConfig
 	}
 	return registryConfig, nil
 }
 
+func ProtoSwiftConfigToSwiftRegistryConfig(protoSwiftConfig *registryv1alpha1.SwiftConfig) (*bufpluginconfig.SwiftRegistryConfig, error) {
+	swiftConfig := &bufpluginconfig.SwiftRegistryConfig{}
+	runtimeLibs := protoSwiftConfig.GetRuntimeLibraries()
+	if runtimeLibs != nil {
+		swiftConfig.Dependencies = make([]bufpluginconfig.SwiftRegistryDependencyConfig, 0, len(runtimeLibs))
+		for _, runtimeLib := range runtimeLibs {
+			dependencyConfig := bufpluginconfig.SwiftRegistryDependencyConfig{
+				Package:  runtimeLib.GetPackage(),
+				Version:  runtimeLib.GetVersion(),
+				Products: runtimeLib.GetProducts(),
+			}
+			platforms := runtimeLib.GetPlatforms()
+			for _, platform := range platforms {
+				switch platform.GetName() {
+				case registryv1alpha1.SwiftPlatformType_SWIFT_PLATFORM_TYPE_MACOS:
+					dependencyConfig.Platforms.MacOS = platform.GetVersion()
+				case registryv1alpha1.SwiftPlatformType_SWIFT_PLATFORM_TYPE_IOS:
+					dependencyConfig.Platforms.IOS = platform.GetVersion()
+				case registryv1alpha1.SwiftPlatformType_SWIFT_PLATFORM_TYPE_TVOS:
+					dependencyConfig.Platforms.TVOS = platform.GetVersion()
+				case registryv1alpha1.SwiftPlatformType_SWIFT_PLATFORM_TYPE_WATCHOS:
+					dependencyConfig.Platforms.WatchOS = platform.GetVersion()
+				default:
+					return nil, fmt.Errorf("unknown platform type: %v", platform.GetName())
+				}
+			}
+			swiftConfig.Dependencies = append(swiftConfig.Dependencies, dependencyConfig)
+		}
+	}
+	return swiftConfig, nil
+}
+
+func SwiftRegistryConfigToProtoSwiftConfig(swiftConfig *bufpluginconfig.SwiftRegistryConfig) *registryv1alpha1.SwiftConfig {
+	protoSwiftConfig := &registryv1alpha1.SwiftConfig{}
+	if swiftConfig.Dependencies != nil {
+		protoSwiftConfig.RuntimeLibraries = make([]*registryv1alpha1.SwiftConfig_RuntimeLibrary, 0, len(swiftConfig.Dependencies))
+		for _, dependency := range swiftConfig.Dependencies {
+			protoSwiftConfig.RuntimeLibraries = append(protoSwiftConfig.RuntimeLibraries, &registryv1alpha1.SwiftConfig_RuntimeLibrary{
+				Package:  dependency.Package,
+				Version:  dependency.Version,
+				Products: dependency.Products,
+				Platforms: []*registryv1alpha1.SwiftConfig_RuntimeLibrary_Platform{
```

Comment:
This doesn't feel 1-1 with the conversion from a protobuf type. Don't we need to check each of the dependency.Platforms.* values and only add a platform entry if the version is non-empty?